torchrl/envs/utils.py

```diff
@@ -49,6 +50,9 @@ def step_mdp(
     exclude_reward: bool = True,
     exclude_done: bool = False,
     exclude_action: bool = True,
+    reward_key: NestedKey = "reward",
+    done_key: NestedKey = "done",
+    action_key: NestedKey = "action",
```

Comment:
@vmoens Would it be possible to merge a version of step_mdp that works with nested key before this whole PR? the fact that this function does not currently support nested is causing all sorts of issues with collectors and so on

Reply:
Sure makes sense
I believe this PR can be split in several smaller ones, let's see what we can do